### ISSUE TITLE
segmentation fault (possibly in kmeans)

### ISSUE THREAD

**Issue by xiaohan2012** [mwydmuch/napkinXC#17]
Hi,

The following snippet gives segmentation fault on my machine:

```python
from napkinxc.models import PLT
from napkinxc.datasets import load_dataset

trn_X, trn_Y = load_dataset('eurlex-4k', "train", verbose=1)
model = PLT('output/test', tree_type='hierarchicalKmeans',
            arity=32,
            seed=25,
            threads=4, verbose=1)
model.fit(trn_X, trn_Y)
```

The output is:

```
napkinXC 0.5.1 - train
  Model: output/test
    Type: plt
  Base models optimizer: liblinear
    Solver: L2R_LR_DUAL, eps: 0.1, cost: 10, max iter: 100, weights threshold: 0.1
  Tree type: hierarchicalKmeans, arity: 32, k-means eps: 0.0001, balanced: 1, weighted features: 0, max leaves: 100
  Threads: 4, memory limit: ~29G
  Seed: 25
Building tree ...
Computing labels' features matrix in 4 threads ...
Hierarchical K-Means clustering in 4 threads ...
[2]    20767 segmentation fault (core dumped)  python mwe.py
```

Would it be possible to fix this?

Other info:

- `napkinXC 0.5.1`
- `Python 3.8.3`
- `Ubuntu 16.04.7 LTS`

**Comment by mwydmuch** [comment]
Hi @xiaohan2012, thank you for reporting the problem! I've managed to replicate it, it seems that for this arity = 32, it fails only for seed = 25, if you change a seed to some other number it works correctly. It seems that this seed generates some kind of edge case that causes the segfault, I will investigate and fix it soon.

**Comment by mwydmuch** [comment]
Closing, because the issue was fixed some time ago.
